dev-tools/xtask/src/a4x2_deploy.rs

```diff
@@ -0,0 +1,472 @@
+    // - stall
+    //
+    // Stalling is why we do a timeout, and we need to handle timeout both at
+    // the TCP level and the HTTP level. `curl -m 5` will hard-out after 5
+    // seconds no matter what's going on.
+    //
+    // That 5 seconds is arbitrary, but it's been working well over in
+    // rackletteadm, from which this logic is copied.
+    //
+    // We could replace curl with rust code, but make sure we do the same thing!
+    while retries > 0 && cmd!(sh, "curl -s -m 5 {api_url}").run().is_err() {
+        retries -= 1;
+        thread::sleep(time::Duration::from_secs(25));
+
+        if retries % 5 == 0 {
+            // I see no reason to error while printing the date
+            let _ = cmd!(sh, "date").run();
+        }
+    }
+
+    if retries == 0 {
+        bail!("timed out waiting for control plane");
+    }
+
+    Ok(())
+}
+
+fn run_tests(sh: &Shell, env: &Environment, args: &RunTestsArgs) -> Result<()> {
+    // TODO: support running end to end tests
+    if args.live_tests {
+        cmd!(sh, "banner 'live tests'").run()?;
+        let _popdir = sh.push_dir(&env.a4x2_dir);
+        let g0ip = get_node_ip(&sh, env, "g0")?;
+
+        let ssh_host = format!("root@{g0ip}");
+        let mut ssh_args_owned = vec!["-i", "../a4x2-ssh-key"];
+        ssh_args_owned.extend_from_slice(&INSECURE_SSH_ARGS);
+        let ssh_args = &ssh_args_owned;
+
+        cmd!(sh, "scp {ssh_args...} live-tests-bundle.tgz {ssh_host}:/zone/oxz_switch/root/root").run()?;
+
+        // If you want any change in functionality for the test runner, update
+        // run-live-tests over in a4x2_package.rs. Don't add it here!
+        let switch_zone_script = r#"
+            set -euxo pipefail
+            tar xvzf live-tests-bundle.tgz
+            cd live-tests-bundle
+            ./run-live-tests
+        "#;
+
+        let remote_script =
+            format!("zlogin oxz_switch bash -c '{switch_zone_script}'");
+
+        // Will error if the live tests fail. This is desired.
+        cmd!(sh, "ssh {ssh_args...} {ssh_host} {remote_script}").run()?;
+    }
+
+    Ok(())
+}
+
+fn teardown_a4x2(sh: &Shell, env: &Environment) -> Result<()> {
+    let _popdir = sh.push_dir(&env.a4x2_dir);
+
+    // destroy a4x2 stuff
+    cmd!(sh, "pfexec ./a4x2 destroy").run()?;
+
+    // destroy the route we added (and any stale ones laying around) Each time
+    // we run `route get` we will get one gateway. If multiple routes have been
+    // added for this IP (by, say, multiple spinup commands without teardowns in
+    // between), then we will need to do this multiple times to fully tear down
+    // all of them. But, I don't like unbounded loops, so we will do at max 10,
+    // which is a number that seems unlikely enough to reach, to me.
+    for _ in 0..10 {
+        let mut route_cmd = cmd!(sh, "pfexec route get 198.51.100.0/24");
+
+        // We get an error code when there is no route (and thus, nothing for
+        // us to delete!)
```

Comment:
I think before ignoring the result here I would make sure the output ends in **not in table**, otherwise it's presumably an error we're _not_ expecting and we'd like to display it to the user and abort.